katdal/visdatav4.py

```diff
@@ -277,24 +279,42 @@ def __init__(self, source, ref_ant='', time_offset=0.0, applycal='',
         self.sensor['Observation/label'] = label
         self.sensor['Observation/compscan_index'] = CategoricalData(list(range(len(label))),
                                                                     label.events)
-        # Use the target sensor of reference antenna to set target for each scan
-        target = self.sensor.get(self.ref_ant + '_target')
-        # Remove initial blank target (typically because antenna starts stopped)
-        if len(target) > 1 and target[0] == 'Nothing, special':
-            target.events, target.indices = target.events[1:], target.indices[1:]
-            target.events[0] = 0
+        # Use the relevant target sensor to set target for each scan
+        ref_target = self.ref_ant or 'cbf'
+        target = self.sensor.get(ref_target + '_target')
         # Move target events onto the nearest scan start
         # ASSUMPTION: Number of targets <= number of scans
         # (i.e. only a single target allowed per scan)
         target.align(scan.events)
+        # Remove initial target if antennas start in mode STOP
+        # (typically left over from previous capture block)
+        for segment, scan_state in scan.segments():
+            # Keep going until first non-STOP scan or a new target is set
+            if scan_state == 'stop' and target[segment.start] is target[0]:
+                continue
+            # Only remove initial target event if we move to a different target
+            if target[segment.start] is not target[0]:
+                target.events = target.events[1:]
+                target.indices = target.indices[1:]
+                target.events[0] = 0
+                target.align(target.events)
+            break
         self.sensor['Observation/target'] = target
         self.sensor['Observation/target_index'] = CategoricalData(target.indices,
                                                                   target.events)
-        # Set up catalogue containing all targets in file, with reference
-        # antenna as default antenna
+        # Set up catalogue containing all targets in file
         self.catalogue.add(target.unique_values)
-        ref_sensor = 'Antennas/%s/antenna' % (self.ref_ant,)
-        self.catalogue.antenna = self.sensor.get(ref_sensor)[0]
+        if self.ref_ant in cam_ants:
+            # If provided, use reference antenna as default antenna
+            cat_ant = [ant for ant in ants if ant.name == self.ref_ant][0]
+        else:
+            # Extract array reference as an antenna object for the catalogue
+            cat_ant = katpoint.Antenna(ants[0].description)
+            cat_ant.name = 'array'
+            cat_ant.delay_model.set(None)
+            cat_ant.pointing_model.set(None)
+            cat_ant = katpoint.Antenna(cat_ant.description)
```

Comment:
This definitely needs an explanation, since it looks like a no-op, although I see it somehow fixes up position_enu. Also the `.description` can be dropped since `Antenna.__init__` can take an Antenna.

When I've created an array reference antenna (in several codebases I think) I've missed out on clearing the delay and pointing models. I'd suggest moving this code into katpoint itself so that it can be shared (it would be a useful step towards an array-wide UVW function).

Reply:
The problem with katpoint.Antennas is that they are not meant to be modified in-place, but rather constructed from scratch to ensure the internals are consistent. From the `katpoint.Antenna` docstring:

{quote}
    It is a bad idea to edit the coordinates of the antenna in-place, as the
    various position tuples will not be updated - reconstruct a new antenna
    object instead.
{quote}

I definitely want to move this to katpoint for the same reason you mention.

Reply:
Document this better.